sdk/aqueduct/integrations/connect_config.py

```diff
@@ -0,0 +1,150 @@
+from enum import Enum
+from typing import Dict, Optional, Union
+
+from aqueduct.constants.enums import MetaEnum, ServiceType
+from aqueduct.error import InternalAqueductError
+from pydantic import BaseModel, Extra, Field
+
+"""Copied mostly over from `aqueduct_executor/operators/connectors/data/config.py` for now, please keep them in sync."""
+
+
+class BaseConnectionConfig(BaseModel):
+    """
+    BaseConfig defines the Pydantic Config shared by all connector Config's, e.g.
+    postgres.Config, mysql.Config, etc.
+    """
+
+    class Config:
+        extra = Extra.forbid
+
+
+class BigQueryConfig(BaseConnectionConfig):
+    project_id: str
+    service_account_credentials: str
+
+
+class MySQLConfig(BaseConnectionConfig):
+    username: str
+    password: str
+    database: str
+    host: str
+    port: str
+
+
+class MongoDBConfig(BaseConnectionConfig):
+    auth_uri: str
+    database: str
+
+
+class PostgresConfig(BaseConnectionConfig):
+    username: str
+    password: str
+    database: str
+    host: str
+    port: Optional[str] = "5432"
+
+
+class AWSCredentialType(str, Enum, metaclass=MetaEnum):
+    ACCESS_KEY = "access_key"
+    CONFIG_FILE_PATH = "config_file_path"
+    CONFIG_FILE_CONTENT = "config_file_content"
+
+
+class S3Config(BaseConnectionConfig):
+    # default type to ACCESS_KEY mainly for backward compatibility
+    type: AWSCredentialType = AWSCredentialType.ACCESS_KEY
+
+    # Access key credentials
+    access_key_id: str = ""
+    secret_access_key: str = ""
+
+    # Config credentials
+    config_file_path: str = ""
+    config_file_content: str = ""
+    config_file_profile: str = ""
+
+    bucket: str = ""
+
+    region: str = ""
+    use_as_storage: str = ""
+
+
+class AthenaConfig(BaseConnectionConfig):
+    # default type to ACCESS_KEY mainly for backward compatibility
+    type: AWSCredentialType = AWSCredentialType.ACCESS_KEY
+
+    # Access key credentials
+    access_key_id: str = ""
+    secret_access_key: str = ""
+    region: str = ""
+
+    # Config credentials
+    config_file_path: str = ""
+    config_file_content: str = ""
+    config_file_profile: str = ""
+
+    database: str = ""
+    output_location: str = ""
+
+
+class SnowflakeConfig(BaseConnectionConfig):
+    username: str
+    password: str
+    account_identifier: str
+    database: str
+    warehouse: str
+    db_schema: Optional[str] = Field("public", alias="schema")  # schema is a Pydantic keyword
+
+    class Config:
+        # Ensures that Pydantic parses JSON keys named "schema" or "db_schema" to
+        # the `db_schema` field
+        allow_population_by_field_name = True
+
+
+class SqlServerConfig(BaseConnectionConfig):
+    username: str
+    password: str
+    database: str
+    host: str
+    port: str
+
+
+class SQLiteConfig(BaseConnectionConfig):
+    database: str
+
+
+IntegrationConfig = Union[
```

Comment:
Are we allowing users to connect compute integrations (K8s, Lambda, etc.) through the SDK?

Reply:
yeah I think so - I figured we could add them here as we go.